pymoab/range.pyx

```diff
@@ -35,4 +35,7 @@ cdef class Range(object):
 
     def __getitem__(self, int index):
         cdef moab.EntityHandle rtn = deref(self.inst)[index]
-        return rtn
+        if index < self.size():
+            return rtn
+        else:
+            raise StopIteration
```

Comment:
The getitem() feature seems to be a deprecated / archaic feature of Python
from before there were iterators:
http://stackoverflow.com/questions/926574/why-does-defining-getitem-on-a-class-make-it-iterable-in-python.
I didn't know/remember this for what it is worth :)

On Wed, Jan 13, 2016 at 7:12 PM Patrick Shriwise <notifications@github.com>
wrote:

> In pymoab/range.pyx
> <https://github.com/scopatz/pymoab/pull/14#discussion_r49671624>:
>
> > @@ -35,4 +35,7 @@ cdef class Range(object):
> >
> >      def __getitem__(self, int index):
> >          cdef moab.EntityHandle rtn = deref(self.inst)[index]
> > -        return rtn
> > +        if index < self.size():
> > +            return rtn
> > +        else:
> > +            raise StopIteration
>
> @scopatz <https://github.com/scopatz> can do. Thanks!
>
> I'm not very familiar with how iterables are supposed to work. I'm
> assuming that if the object doesn't have an *iter* method then it will
> try to use *getitem* method to do the same thing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scopatz/pymoab/pull/14/files#r49671624>.
>
-- 

Asst. Prof. Anthony Scopatz
Nuclear Engineering Program
Mechanical Engineering Dept.
University of South Carolina
scopatz@cec.sc.edu
Office: (803) 777-7629
Cell: (512) 827-8239
Check my calendar
<https://www.google.com/calendar/embed?src=scopatz%40gmail.com>